### ISSUE TITLE
First character is skipped on logging message

### ISSUE THREAD

**Issue by ghost** [microsoft/vscode-cpptools#954]
Hi,
I'm using the debugger in remote. It worked quite well, except for one thing: the first character is skipped for each logged message, this is the script I'm using to debug a remote program on CentOs:
```
    {
      "name": "C++ remote launch (ssh)",
      "type": "cppdbg",
      "request": "launch",
      "program": "/home/renaud/workspace/LGDN_001_DEMONSTRATOR/testApplications/lgdn_001_demonstrator/Lgdn001Demonstrator", // Executable on the remote machine
      "args": ["-v2", "-j5", "-t3", "13"],
      "stopAtEntry": false,
      "cwd": "/home/renaud/workspace/LGDN_001_DEMONSTRATOR/testApplications/lgdn_001_demonstrator", // Directory on remote machine
      "environment": [],
      "externalConsole": true,
      "sourceFileMap": {
        "/home/renaud/workspace/LGDN_001_DEMONSTRATOR/testApplications/lgdn_001_demonstrator/src": "${workspaceRoot}/software/src" // "Compile time source location": "local source location"
      },
      "pipeTransport": {
        "pipeCwd": "/usr/bin", // Local directory of pipe program
        "pipeProgram": "ssh", // Pipe program
        "pipeArgs": ["renaud@10.1.38.10"],
        "debuggerPath": "/usr/bin/gdb" // Debugger location on the remote machine
      },
      "logging": {
        "engineLogging": true
      }
    }
```

An example of message expected:
> 0min 1sec: [3;34mDEBUG: Lockable: Unlock the instance of the program

But I'm getting that:
> min 1sec: [3;34mDEBUG: Lockable: Unlock the instance of the program

With the engine logging enabled, I get that information (hope it can help):
> 1: (5826) ->0min 1sec: [3;34mDEBUG: Lockable: Unlock the instance of the program

Also everything is logged in the debug console, and not on an external terminal (as requested in the script), but that's not problem for me.

If you want more details, just ask.

**Comment by pieandcakes** [comment]
@Tankypon how are you logging these messages?

**Comment by ghost** [comment]
I logged these messages on the standard console output with std::cout. Also the sequence '[3;34m' should be understandable by the console to specify the color font. At least it was working on the default console on Ubuntu (not the console of VSCode).

**Comment by pieandcakes** [comment]
@Tankypon You are seeing this in the debug console within VSCode correct? I can see if I can duplicate the missing character but the color portion would be VS Code's issue to fix as they control the debug console and we just output to it.

**Comment by ghost** [comment]
@pieandcakes This is correct. The code is executed on a remote machine (on CentOS) and the output is displayed in the debug console within VSCode (on a Ubuntu machine).

**Comment by Iansa9** [comment]
I am no expert, but maybe you are dealing with CRLF issues between Operating Systems, I had a similar problem with a simple c++ program and that was the issue.
